### DISCUSSION TITLE
Can `auto_session_suppress_dirs` and `auto_session_allowed_dirs` be used to whitelist directories?

### DISCUSSION THREAD

**Discussion by adrian5** [rmagatti/auto-session#254] [Q&A] [answered]
Is that possible? Something like `"*"` for `suppress` and then enabling a few paths in `allowed`? Or does that go against the design of the plugin?

I'm thinking I'd prefer to only store sessions for designated paths I'm regularly working in, with multiple open buffers. For quick edits somewhere in the file system I don't want to record a session.

**Comment by rmagatti** [comment, marked as answer]
Right now that's not possible, as suppressing takes precedence over allowing.

**Reply by adrian5** [reply]
I think I might have misunderstood these options. Does `auto_session_allowed_dirs` already behave like a whitelist on its own? Namely, as soon as it's being set, _only_ dirs listed there qualify for session management? That part wasn't clear to me.

**Reply by rmagatti** [reply]
You are correct, it does in fact behave like an allow list. Sorry, I should have mentioned this in your first question. That should work for what you want to do.
